crates/nu-cli/tests/commands/parse.rs

```diff
@@ -2,31 +2,67 @@ use nu_test_support::fs::Stub;
 use nu_test_support::playground::Playground;
 use nu_test_support::{nu, pipeline};
 
-#[test]
-fn extracts_fields_from_the_given_the_pattern() {
-    Playground::setup("parse_test_1", |dirs, sandbox| {
-        sandbox.with_files(vec![Stub::FileWithContentToBeTrimmed(
-            "key_value_separated_arepa_ingredients.txt",
-            r#"
-                VAR1=Cheese
-                VAR2=JonathanParsed
-                VAR3=NushellSecretIngredient
-            "#,
-        )]);
+mod simple {
+    use super::*;
 
-        let actual = nu!(
-            cwd: dirs.test(), pipeline(
-            r#"
-                open key_value_separated_arepa_ingredients.txt
-                | parse "{Name}={Value}"
-                | nth 1
-                | get Value
-                | echo $it
+    #[test]
+    fn extracts_fields_from_the_given_the_pattern() {
+        Playground::setup("parse_test_1", |dirs, sandbox| {
+            sandbox.with_files(vec![Stub::FileWithContentToBeTrimmed(
+                "key_value_separated_arepa_ingredients.txt",
+                r#"
+                    VAR1=Cheese
+                    VAR2=JonathanParsed
+                    VAR3=NushellSecretIngredient
+                "#,
+            )]);
+
+            let actual = nu!(
+                cwd: dirs.test(), pipeline(
+                r#"
+                    open key_value_separated_arepa_ingredients.txt
+                    | lines
+                    | each { echo $it | parse "{Name}={Value}" }
```

Comment:
So this is the one interesting impact of my change: since we have the start/end of input anchors, we need to split the lines up to make this test work. I think previously it would try to find all non-overlapping matches. This _is_ a change in behaviour, but I think it's good.

If we did want the previous behaviour, I think it would be good to put it behind a flag.

Reply:
I guess the one inconsistency with this behaviour is that `--regex` would find all non overlapping matches (assuming one doesn't anchor it in a similar way). Maybe we could come back later and make this a little more consistent by adding a flag?

Reply:
Sounds good. I'm okay landing this and filing a follow-up issue for the consistency bit you mention.

Reply:
I'll file an issue for it. Would be good to discuss which approach is preferred 👍